config/src/config.rs

```diff
@@ -150,6 +150,9 @@ pub struct Connections {
     /// peers should bind to
     pub server_addr: SocketAddr,
 
+    /// Public address
+    pub public_addr: SocketAddr,
```

Comment:
Does it make sense to make this an `Option`? For people that don't have a public IP.

```suggestion
    pub public_addr: Option<SocketAddr>,
```

Reply:
They only should have to put the same IP that in server_address

Reply:
> They only should have to put the same IP that in server_address

For it to work, it is true that leaving the default address `0.0.0.0:...` just does the job as you still have to put some IP in the message that is exchanged. 

I am not claiming though that that's ideal in the long term. This is just the option with the lowest impact (it doesn't cause a hard fork as the protobuf schema remains intact).